nighthawk/source/client/utility.cc

```diff
@@ -0,0 +1,64 @@
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_split.h"
+
+#include "common/http/utility.h"
+#include "common/network/utility.h"
+
+#include "nighthawk/source/common/utility.h"
+
+namespace Nighthawk {
+
+namespace PlatformUtils {
+
+// returns 0 on failure. returns the number of HW CPU's
+// that the current thread has affinity with.
+// TODO(oschaaf): mull over what to do w/regard to hyperthreading.
+uint32_t determineCpuCoresWithAffinity() {
+  uint32_t concurrency = 0;
+  int i;
+  pthread_t thread = pthread_self();
+  cpu_set_t cpuset;
+  CPU_ZERO(&cpuset);
+  i = pthread_getaffinity_np(thread, sizeof(cpu_set_t), &cpuset);
```

Comment:
One thing we should decide on early is if we want an OS syscall abstraction layer. It's pretty nice for when you want to mock, and isn't _that_ much additional boilerplate to use. See https://github.com/envoyproxy/envoy/blob/master/include/envoy/api/os_sys_calls.h and its uses in  Envoy.